PEP 360: Point to the Experts Index (#2276)
python/peps: pep-0360.txt

Created: 30-May-2006
Post-History:

> .. note:: The `experts index <https://devguide.python.org/experts>`__ in the
>           Python developer's guide obsoletes this PEP.

Reviewer: ```suggestion
.. note:: The modules listed here are now maintained in the standard library
          and no new ones will be added, so this PEP is of historical interest
          only. See the `Experts Index <https://devguide.python.org/experts>`__
          for an up-to-date listing of modules and their associated experts.
```

Reviewer 2: This is not quite true, expat is still maintained externally.

Author: ```suggestion
.. note:: This PEP is obsolete, as new non-provisional standard library modules are maintained internally. See the `place to be determined` for a list of remaining externally managed modules.
```

Not word wrapped. Is this more in keeping with what you were thinking of? Intentionally didn't specify where the "list of remaining external modules" will be, so that we can discuss the wording apart from the proposed location.

A

Reviewer: > This is not quite true, expat is still maintained externally.

True, `libexpat` is maintained externally, but technically it is a C library (as clarified in the existing PEP text), not a standard library module (as referred to in the note), just like e.g. the `zlib` library for the `zlib` module, OpenSSL for the `ssl` module, and others. Here's a revised suggestion making that more clear.

```suggestion
.. note:: The Python modules listed below (not including the Expat C library)
          are now maintained in the standard library and no new ones
          will be added here, so this PEP is of historical interest only.
          See the `Experts Index <https://devguide.python.org/experts>`__
          for an up-to-date listing of modules and their associated experts.
```

> Is this more in keeping with what you were thinking of? Intentionally didn't specify where the "list of remaining external modules" will be, so that we can discuss the wording apart from the proposed location.

Not exactly (again, sorry)—as Guido mentions, its not completely true, since several recently added modules are maintained separately, and I really don't think we should block clarifying that the PEP is unmaintained and obselete on having to create a replacement, which is a non-trivial task (especially if we include third-party libraries wrapped by standard library modules, which will require a fair amount of work to track down).

Reviewer: @JelleZijlstra @AA-Turner @gvanrossum Are we okay with submitting the above revision to the SC, or do you suggest making further changes? Or, what specifically do we want to ask the SC to approve/opine on here?

Author: > Not exactly (again, sorry)—as Guido mentions, its not completely true, since several recently added modules are maintained separately, and I really don't think we should block clarifying that the PEP is unmaintained and "of historical interest only" on having to create a replacement, which is a non-trivial task (especially if we include third-party libraries wrapped by standard library modules, which will require a fair amount of work to track down).

Hmm OK. My challenge then is why the "see the Experts..." is relevent, if we're not updating it to contain externally maintained modules. A simpler / smaller change would be:

```suggestion
.. note:: This PEP is for historical interest only, and does not reflect the current state of externally maintained modules.
```

It does feel odd to remove this though as the supposed listing whilst not creating a new home, although I note the scope widening significantly for little marginal benefit.

A

Reviewer: ```suggestion
.. note:: This PEP is of historical interest only, and does not reflect the
          current state of externally-maintained modules.
```

Add line breaks, hyphen and fix word ("for" -> "of").

> Hmm OK. My challenge then is why the "see the Experts..." is relevent, if we're not updating it to contain externally maintained modules. A simpler / smaller change would be:

I was quoting you :rofl: In all seriousness, fair point; I was considering removing it but kept it from your original since it does list the current maintainers of the modules listed in the PEP. But that's not terribly important.

> It does feel odd to remove this though as the supposed listing whilst not creating a new home, although I note the scope widening significantly for little marginal benefit.

Yeah, though it doesn't actually "remove" anything, it just makes the current status of the PEP clear. I agree that its not all that important either way, to be honest—the main source of my confusion was it being linked in the devguide, and python/devguide#802 fixes that.
